### DISCUSSION TITLE
How can we import everything (*) from an external task file when using blocks?

### DISCUSSION THREAD

**Discussion by brendangilmore** [pybricks/support#1893] [Q&A]
As an FLL coach moving our team to PyBricks, we're hoping to setup a method to help the varying levels of skills on our team work together in coding. Our more advanced members will use python, implementing a core robot library to provide all the various functions. Then individual members can work on mission ideas using either Python or blocks. We'd rather not require the missions to import each and every function from the robot library. A Python mission file would simply:
`from our_robot import *`
The trouble is we can't figure out a way to do that using blocks. When we try supplying * in the External Task setup block, it replaces it with _2 in the equivalent python.

**Comment by laurensvalk** [comment]
> As an FLL coach moving our team to PyBricks, we're hoping to setup a method to help the varying levels of skills on our team work together in coding. 

This is great! I'm glad you discovered the import functionality.

> The trouble is we can't figure out a way to do that using blocks. When we try supplying * in the External Task setup block, it replaces it with _2 in the equivalent python.

With `*`, the block editor wouldn't know which blocks to make available. However, there is another way that achieves essentially the same, as follows:

---------

The block users have to import just one function. And then they would specify which function they really want to call, by name:

![image](https://github.com/user-attachments/assets/db396a11-713b-425a-bbfd-8f4949e72a11)

Create a file called `team_python.py` with the following contents:

```python
from pybricks.tools import wait

# Example function
def hello(text):
    print("Hi,", text)

# Another example function
def beep(hub, count):
    for i in range(count):
        hub.speaker.beep()
        wait(50)

# This helper function will select one of the above functions
# and call it by passing on any of the arguments you give.
def function(name, *args, **kwargs):
    return globals()[name](*args, **kwargs)
```

So block users still achieve a single import. The beep example function illustrates how Python functions can make use of the devices that are set up by the blocks.

If you are going to use multitasking, this will look a bit different. Let me know if you'd like an example for that too.

**Comment by BertLindeman** [comment]
The blocks have been moved into the category "variables".

Currently the blocks program looks like this:

<img width="937" height="650" alt="image" src="https://github.com/user-attachments/assets/de78017b-cbe2-48ea-a14f-86c9f6fcc4e3" />
